app/Controller.cpp

```diff
@@ -80,9 +85,16 @@ void Controller::controlLoop() {
     // update next target time
     next_loop_time += duration;
 
-    //
-    // @TODO groundbreaking robotics goes here
-    //
+    double speed_error, heading_error, command_throttle, command_steering, actual_throttle, actual_steering;
+    double steeringVel = 0.0; // TODO: Understanding the current_steering_vel and update accordingly
+
+    this->model_->getError(speed_error, heading_error);
+    command_throttle = this->pid_speed_->getCommand(speed_error);
+    command_steering = this->pid_heading_->getCommand(heading_error);
+  
+    this->model_->limits_->limit(command_throttle, command_steering, steeringVel, actual_throttle, actual_steering, steeringVel, duration);
```

Comment:
I believe (@selyard can confirm) the limiting is done internally by the `model` class under Spencer's new setup.

Reply:
confirmed as plan, still to be implemented though